packages/dappmanager/src/modules/chains/drivers/ethereum.ts

```diff
@@ -23,14 +24,16 @@ const gethSyncHelpUrl = whyDoesGethTakesSoMuchToSync;
  * }
  */
 export async function ethereum(
-  dnp: InstalledPackageData
+  dnp: InstalledPackageData,
+  chainDriver: ChainDriverSpecs
 ): Promise<ChainDataResult> {
   const container = dnp.containers[0];
   if (!container) throw Error("no container");
   const containerDomain = getPrivateNetworkAlias(container);
 
+  const port = chainDriver.portNumber || 8545;
   // http://ropsten.dappnode:8545
-  const apiUrl = `http://${containerDomain}:8545`;
+  const apiUrl = `http://${containerDomain}:${port}`;
```

Comment:
Here you are using the `containerDomain` obtained from the first container which is not ideal. Instead, you should try to use first the `serviceName` from the `chainDriver`, and if is not defined, then use the other approach. Take as reference https://github.com/dappnode/DNP_DAPPMANAGER/blob/8eaaff5ddf0ffacac58ba9ef1f4cb5ad97d26676/packages/dappmanager/src/modules/chains/drivers/ethereum2.ts#L17

Reply:
Looking into it, thanks for the review

Reply:
Done, tested it with and without chain declaration in package manifests.